File: backend/models/InputInventory.js
const mongoose = require("mongoose");

const inputInventorySchema = new mongoose.Schema(
  {
    crop: {
      type: String,
      required: true,
      index: true,
      trim: true,
    },
    inputName: {
      type: String,
      required: true,
      index: true,
      trim: true,
    },
    qtyPerAcre: {
      type: Number,
      required: true,
      min: 0,
    },
    unit: {
      type: String,
      required: true,
      trim: true,
    },
    sellerName: {
      type: String,
      required: true,
      trim: true,
    },
    district: {
      type: String,
      required: true,
      index: true,
      trim: true,
    },
    state: {
      type: String,
      required: true,
      index: true,
      trim: true,
    },
    distanceKm: {
      type: Number,
      required: true,
      min: 0,
    },
    pricePerUnit: {
      type: Number,
      required: true,
      min: 0,
    },
    stockQty: {
      type: Number,
      required: true,
      min: 0,
    },
    phone: {
      type: String,
      default: "",
      trim: true,
    },
    source: {
      type: String,
      enum: ["seed", "seller"],
      default: "seed",
    },
  },
  { timestamps: true }
);

inputInventorySchema.index({ crop: 1, inputName: 1, district: 1, state: 1 });

module.exports = mongoose.model("InputInventory", inputInventorySchema);

File: backend-java/src/main/java/com/agrisense/backend/model/InputInventory.java
package com.agrisense.backend.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.index.CompoundIndex;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.time.LocalDateTime;

/**
 * Input inventory document — mirrors models/InputInventory.js
 * Collection: "inputinventories"
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "inputinventories")
@CompoundIndex(def = "{'crop': 1, 'inputName': 1, 'district': 1, 'state': 1}")
public class InputInventory {

    @Id
    private String id;

    @Indexed
    @Field("crop")
    private String crop;

    @Indexed
    @Field("inputName")
    private String inputName;

    @Field("displayName")
    private String displayName;

    @Field("qtyPerAcre")
    private Double qtyPerAcre;

    @Field("unit")
    private String unit;

    @Field("sellerName")
    private String sellerName;

    @Indexed
    @Field("district")
    private String district;

    @Indexed
    @Field("state")
    private String state;

    @Field("distanceKm")
    private Double distanceKm;

    @Field("pricePerUnit")
    private Double pricePerUnit;

    @Field("stockQty")
    private Double stockQty;

    @Field("phone")
    private String phone = "";

    @Field("source")
    private String source = "seed"; // "seed" | "seller"

    @CreatedDate
    private LocalDateTime createdAt;

    @LastModifiedDate
    private LocalDateTime updatedAt;
}
